examples/runc-hook/main.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+	ocispec "github.com/opencontainers/runtime-spec/specs-go"
+	api "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/k8sutil"
+	"github.com/kinvolk/inspektor-gadget/pkg/runcfanotify"
+)
+
+var (
+	outputList = flag.String("output", "add,remove", "comma-separated list of events to print [add,remove,config]")
+
+	outputAdd    bool
+	outputRemove bool
+	outputConfig bool
+
+	hookPreStart = flag.String("prestart", "", "command to run in the PreStart hook")
+	hookPostStop = flag.String("poststop", "", "command to run in the PostStop hook")
+	env          = flag.String("env", "", "the environ")
+	dir          = flag.String("dir", "", "dir")
+	timeout      = flag.String("timeout", "10s", "timeout")
+
+	publishKubernetesEvent = flag.Bool("publish-kubernetes-event", false, "Publish an event using the Kubernetes Event API")
+	kubeconfig             = flag.String("kubeconfig", "", "kubeconfig")
+	node                   = flag.String("node", "", "Node name")
+
+	notifier *runcfanotify.RuncNotifier
+	client   *kubernetes.Clientset
+)
+
+func publishEvent(reason, message string) {
+	eventTime := metav1.NewTime(time.Now())
+	event := &api.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%v.%x", *node, time.Now().UnixNano()),
+			Namespace: "default",
+		},
+		Source: api.EventSource{
+			Component: "RuncHook",
+			Host:      *node,
+		},
+		Count:               1,
+		ReportingController: "github.com/kinvolk/inspektor-gadget",
+		ReportingInstance:   os.Getenv("POD_NAME"), // pod name
+		FirstTimestamp:      eventTime,
+		LastTimestamp:       eventTime,
+		InvolvedObject: api.ObjectReference{
+			Kind: "Node",
+			Name: *node,
+			// Uncomment to make it visible in 'kubectl describe node'
+			//UID: types.UID(*node),
+		},
+		Type:    api.EventTypeNormal,
+		Reason:  reason,
+		Message: message,
+	}
+
+	if _, err := client.CoreV1().Events("default").Create(context.TODO(), event, metav1.CreateOptions{}); err != nil {
```

Comment:
Any reason to at least not hardcode any timeout in the context, so we don't get stalled if the APIserver is down?

It can be done in follow-up PRs if this is still in the early works, a TODO comment will be great in that case, though :)